### ISSUE TITLE
JSII doesnt genereate nullable enum in .Net

### ISSUE THREAD

**Issue by McDoit** [awslabs/jsii#414]
The following code in the .net CDK

```
  [JsiiByValue]
  public class SubnetSelection : ISubnetSelection
  {
    [JsiiProperty("subnetName", "{\"primitive\":\"string\",\"optional\":true}", true)]
    public string SubnetName { get; set; }

    [JsiiProperty("subnetType", "{\"fqn\":\"@aws-cdk/aws-ec2.SubnetType\",\"optional\":true}", true)]
    public SubnetType SubnetType { get; set; }
  }
```
doesnt follow the nullability in vpc-ref.ts

```
export interface SubnetSelection {
  /**
   * Place the instances in the subnets of the given type
   *
   * At most one of `subnetType` and `subnetName` can be supplied.
   *
   * @default SubnetType.Private
   */
  readonly subnetType?: SubnetType;

  /**
   * Place the instances in the subnets with the given name
   *
   * (This is the name supplied in subnetConfiguration).
   *
   * At most one of `subnetType` and `subnetName` can be supplied.
   *
   * @default name
   */
  readonly subnetName?: string;
}
```

Which makes .Net code throw `Only one of subnetType and subnetName can be supplied` everytime a SubnetName is used, even if no value is defined for SubnetType

https://github.com/awslabs/aws-cdk/blob/master/packages/@aws-cdk/aws-ec2/lib/vpc-ref.ts

**Comment by McDoit** [comment]
Same problem with SslPolicy in ApplicationListenerProps
```
  [JsiiByValue]
  public class ApplicationListenerProps : IApplicationListenerProps, IBaseApplicationListenerProps
  {
    [JsiiProperty("loadBalancer", "{\"fqn\":\"@aws-cdk/aws-elasticloadbalancingv2.IApplicationLoadBalancer\"}", true)]
    public IIApplicationLoadBalancer LoadBalancer { get; set; }

    [JsiiProperty("certificateArns", "{\"collection\":{\"kind\":\"array\",\"elementtype\":{\"primitive\":\"string\"}},\"optional\":true}", true)]
    public string[] CertificateArns { get; set; }

    [JsiiProperty("defaultTargetGroups", "{\"collection\":{\"kind\":\"array\",\"elementtype\":{\"fqn\":\"@aws-cdk/aws-elasticloadbalancingv2.IApplicationTargetGroup\"}},\"optional\":true}", true)]
    public IIApplicationTargetGroup[] DefaultTargetGroups { get; set; }

    [JsiiProperty("open", "{\"primitive\":\"boolean\",\"optional\":true}", true)]
    public bool? Open { get; set; }

    [JsiiProperty("port", "{\"primitive\":\"number\",\"optional\":true}", true)]
    public double? Port { get; set; }

    [JsiiProperty("protocol", "{\"fqn\":\"@aws-cdk/aws-elasticloadbalancingv2.ApplicationProtocol\",\"optional\":true}", true)]
    public ApplicationProtocol Protocol { get; set; }

    [JsiiProperty("sslPolicy", "{\"fqn\":\"@aws-cdk/aws-elasticloadbalancingv2.SslPolicy\",\"optional\":true}", true)]
    public SslPolicy SslPolicy { get; set; }
  }
```

https://github.com/awslabs/aws-cdk/blob/master/packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

**Comment by fulghum** [comment]
@RomainMuller  – I moved this from the CDK board to the (new) .NET board.  It sounds like there's .NET specific work after the jsii core work?

**Comment by RomainMuller** [comment]
Well. There was an issue in the .Net code generator that I have fixed as part of my work in #432. Once this ships, this bug should be gone.
